cv_bridge/test/enumerants.py

```diff
@@ -46,13 +48,19 @@ def test_encode_decode_cv2(self):
                             original = np.uint8(np.random.randint(0, 255, size=(h, w, channels)))
                         rosmsg = cvb_en.cv2_to_imgmsg(original)
                         newimg = cvb_de.imgmsg_to_cv2(rosmsg)
+                        compress_rosmsg = cvb_en.cv2_to_compressed_imgmsg(original ,f)
+                        newimg2         = cvb_de.compressed_imgmsg_to_cv2(compress_rosmsg)
                         self.assert_(original.dtype == newimg.dtype)
+                        self.assert_(original.dtype == newimg2.dtype)
                         if channels == 1:
                             # in that case, a gray image has a shape of size 2
                             self.assert_(original.shape[:2] == newimg.shape[:2])
+                            self.assert_(original.shape[:2] == newimg2.shape[:2])
                         else:
                             self.assert_(original.shape == newimg.shape)
+                            self.assert_(original.shape == newimg2.shape)
```

Comment:
Just have another set of for loops, it will make things easier to track. In that new set, just have the number of channels in ([],1,3,4) (ignore two for jpg).
Also, you are erasing the first set of fmts so that's why you should have a second set (I will fix the tests to use fmts as they don't for now).